Sources/Utilities/Storage/Session/PersistentSessionManager.swift

```diff
@@ -173,12 +196,22 @@ final class PersistentSessionManager: SessionManager {
 
         tokenProvider.clear()
         tokenResponse = nil
-        user = nil
+        user.send(nil)
     }
 
-    func clearAllSessionData() {
+    func clearAllSessionData() throws {
         encryptedStore.deleteItem(itemName: .persistentSessionID)
         unprotectedStore.removeObject(forKey: .returningUser)
         unprotectedStore.removeObject(forKey: .accessTokenExpiry)
+
+        try sessionBoundData.forEach {
+            try $0.delete()
+        }
```

Comment:
Should this be at the top of the function so if wallet data fails to delete we throw the error before deleting the above session data?

Reply:
Yes, makes sense to me 🙂